### ISSUE TITLE
Support for SSL enforcement modes for AlloyDB

### ISSUE THREAD

**Issue by jdub55** [hashicorp/terraform-provider-google#15644]
### Description

AlloyDB now provides an ability to configure various SSL enforcement modes. 
Public docs: https://cloud.google.com/alloydb/docs/instance-ssl

Terraform provider needs to support these new configuration options as well. 

### New or Affected Resource(s)

GoogleCloudPlatform/alloy-db

### Potential Terraform Configuration

TBA

### References

* #0000

b/298050681

**Comment by nancynh** [comment]
I am currently working on this!

**Comment by nancynh** [comment]
Have a pending PR [here](https://github.com/nancynh/magic-modules/commit/4e3c470c56c7afae292bb5d4cffdfedadf880503). Need to wait on some API changes before completing testing and sending it off for review. More details at b/293535381#comment7

**Comment by nancynh** [comment]
This feature had been released in [version 5.3.0](https://github.com/hashicorp/terraform-provider-google/tree/release-5.3.0). We can close this issue now

**Comment by nancynh** [comment]
@roaks3 Could you close this issue? It's been merged into TF for awhile now: https://github.com/GoogleCloudPlatform/terraform-google-conversion/pull/1586
